deploy.sh

```diff
@@ -0,0 +1,51 @@
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [ $branch != dev ]
+then
+    while true; do
+    read -p "You're not on branch dev. Are you sure you want to proceed? [y/n]" yn
+    case $yn in
+        [Yy]* ) break;;
+        [Nn]* ) exit 1;;
+        * ) echo "Please answer y or n.";;
+    esac
+done
+fi
+echo Linting...
+yarn lint
+if [ $? -ne 0 ]
+then
+    echo "Lint failed. Aborting."
+    exit 1
+fi
+echo Building...
+yarn build
+if [ $? -ne 0 ]
+then
+    echo "Build failed. Aborting."
+    exit 1
+fi
+echo Build finished.
+echo Preparing to configure AWS credentials. Get these from someone with access to AWS IAM.
+echo They will likely need to create a new access key. An IAM user with the relevant permissions already exists.
+echo "Contact the web team for more details."
+echo Leave \'Default region name\' and \'Default output format\' blank.
+aws configure
+echo Current contents of bucket:
+aws s3 ls s3://michiganhackers.org
+echo Preparing to synchronizing build.
+while true; do
+    read -p "Are you sure you want to deploy? This cannot be undone. [y/n]" yn
+    case $yn in
+        [Yy]* ) break;;
+        [Nn]* ) exit 1;;
+        * ) echo "Please answer y or n.";;
+    esac
+done
+echo "Synchronizing backup folder..."; 
+aws s3 sync s3://michiganhackers.org s3://michiganhackers.org/backup --exclude "backup"
+echo "Synchronizing s3 bucket with local build folder..."
+aws s3 sync build s3://michiganhackers.org
```

Comment:
is "sync" gonna overwrite the old thing/replace old files with new files? my concern is with filenames that have hashes in them, will the old JS files still be in the bucket after synchronization?

Reply:
They will remain. I can see this becoming a problem. I might just nuke the non-backup objects before I synchronize. Thoughts?

Reply:
@christopherhafke Given that you perform a backup, I'm fine with nuking before sync. 

Additional question: is it possible to **create folders** in the bucket? we could timestamp and store multiple sets of backups and manually delete later. Thoughts?

Reply:
@eashwar Here's my procedure, let me know if you can spot any flaws:
1. Nuke backup
2. Copy current contents into backup
3. Nuke current contents
4. Copy local build into current contents

It should be possible to create timestamped folders, in which case I suppose we wouldn't have to delete any backup folders.

Reply:
Your procedure looks good to me. It was what I imagined before asking about time-stamping backups. I think it's okay to keep just one backup, and any previous backups we need to restore can be done by reverting commits on the repo and deploying. It's your call.

Reply:
Yeah, the backup is really a last-case measure anyway; if something went wrong, I'd personally just build on an old commit rather than copying the backup. I think I'll just go with the simpler procedure for now.